video_xblock/tests/test_backends.py

```diff
@@ -0,0 +1,380 @@
+        """
+        player = self.player[backend]
+        if backend == 'brightcove':
+            self.mocked_objects[backend] = {
+                'obj': brightcove.BrightcoveApiClient,
+                'attrs': ['get', ],
+                'value': [copy.deepcopy(brightcove.BrightcoveApiClient.get), ]
+            }
+            self.mocked_objects[self.xblock.display_name] = {
+                'obj': self.xblock,
+                'attrs': ['metadata', ],
+                'value': [copy.deepcopy(self.xblock.metadata), ]
+            }
+            self.xblock.metadata = BrightcoveDefaultTranscriptsMock(
+                mock_magic=self.xblock.metadata
+            ).no_credentials(event)
+            brightcove.BrightcoveApiClient.get = BrightcoveDefaultTranscriptsMock(
+                mock_magic=brightcove.BrightcoveApiClientError
+            ).api_client_get(event)
+        elif backend == 'youtube':
+            self.mocked_objects[backend] = {
+                'obj': player,
+                'attrs': ['fetch_default_transcripts_languages'],
+                'value': [player.fetch_default_transcripts_languages, ]
+            }
+            player.fetch_default_transcripts_languages = YoutubeDefaultTranscriptsMock()\
+                .fetch_default_transcripts_languages(event)
+        elif backend == 'wistia':
+            self.mocked_objects[backend] = {
+                'obj': requests,
+                'attrs': ['get', ],
+                'value': [copy.deepcopy(requests.get), ]
+            }
+            self.mocked_objects['babelfish'] = {
+                'obj': babelfish,
+                'attrs': ['Language', ],
+                'value': [copy.deepcopy(babelfish.Language), ]
+            }
+            requests.get = WistiaDefaultTranscriptsMock(mock_magic=requests.exceptions.RequestException).get(event)
+        else:
+            # vimeo
+            pass
+
+    @override_settings(ALL_LANGUAGES=ALL_LANGUAGES)
+    @data(
+        *(
+            zip(
+                backends,
+                media_ids,  # video ids
+                [  # mocked events
+                    YoutubeDefaultTranscriptsMock().get_events(),
+                    BrightcoveDefaultTranscriptsMock().get_events(),
+                    WistiaDefaultTranscriptsMock().get_events(),
+                    # for vimeo we add at least one random event to test returned result while no logic present
+                    ['get_transcripts']
+                ],
+                [  # expected results per event
+                    YoutubeDefaultTranscriptsMock().expected_value,
+                    BrightcoveDefaultTranscriptsMock().expected_value,
+                    WistiaDefaultTranscriptsMock().expected_value,
+                    lambda x: ([], '')
+                ]
+            )
+        )
+    )
+    @unpack
+    def test_get_default_transcripts(self, backend, video_id, events, expected_result):
```

Comment:
Here you might swap `expected_result` with `default_trans_mock` so you can call `default_trans_mock.appy_transcripts_mock()` at line 278.